### PR TITLE
Add MemcacheCheck

### PR DESCRIPTION
**Pull request by revathi-murali**
## Summary of change

- Added `MemcacheCheck` that takes in an array of `memached servers` to check the status.
- Added `host` nil check in `HttpEndPointCheck`.

## Reviewers

- [x] @edk

### CODE REVIEW

File: simple_health_check.gemspec [hunk 1]
```diff
@@ -31,6 +31,7 @@ Gem::Specification.new do |spec|
   spec.require_paths = ["lib"]
 
   spec.add_dependency("rails", ">= 5.0")
+  spec.add_dependency("dalli", "2.7.10")
```
Comment (revathi-murali) [inline simple_health_check.gemspec]: Let me know if I had to add `redis` gem here as well as `redis` gem is being used in `RedisCheck`

File: lib/simple_health_check/memcache_check.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+  def initialize service_name: 'memcache', check_proc: nil, config: []
+    @service_name = service_name
+    @proc = check_proc || SimpleHealthCheck::Configuration.memcache_check_proc
+    @type = 'internal'
+    @memcache_servers = config
+  end
+
+  def call(response:)
+    super do
```
Comment (edk) [inline lib/simple_health_check/memcache_check.rb]: oh neat!  i've not seen this pattern before.

**Comment by edk** [comment]
👍

**Comment by revathi-murali** [comment]
[lockit v4.9.2] merged via Lockit CLI on 2019-10-15 17:20:42.727799 +0530 IST m=+4.885071868

**Pull request merged**